### PR TITLE
Fix debug usage of TraceWriters

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
Handle the following:
- tracer.writer can no longer be None
- custom trace writers
- the existing usage of AgentWriter was incorrect

### CODE REVIEW

File: tests/integration/test_debug.py [hunk 1]
```diff
@@ -271,6 +275,28 @@ def test_agentless(monkeypatch):
...
+            pass
+
+        def write(self, spans=None):
+            # type: (Optional[List[Span]]) -> None
+            pass
+
+    tracer.writer = CustomWriter()
+    info = debug.collect(tracer)
+
+    assert info.get("agent_url", "CUSTOM")
```
Comment (Yun-Kim) [inline tests/integration/test_debug.py:297]: Should this be:
```suggestion
    assert info.get("agent_url", None) == "CUSTOM"
```

**Review by Yun-Kim — CHANGES_REQUESTED** [review]
One small nit but otherwise LGTM

File: tests/integration/test_debug.py (hunk 1 above)
Comment (Kyle-Verhoog) [reply tests/integration/test_debug.py:297]: lol I copied this check from the test above which also does the same thing 🤦 . Nice catch!

**Review by Yun-Kim — APPROVED** [review]
😎

File: ddtrace/internal/debug.py [hunk 2]
```diff
@@ -55,6 +47,9 @@ def collect(tracer):
             agent_error = "Agent not reachable at %s. Exception raised: %s" % (agent_url, str(e))
         else:
             agent_error = None
+    else:
+        agent_url = "CUSTOM"
```
Comment (brettlangdon) [inline ddtrace/internal/debug.py:51]: `"UNKNOWN"`?

Not sure if "custom" makes sense here unless someone knows about this check/branch?

File: tests/integration/test_debug.py [hunk 3]
```diff
@@ -268,7 +272,29 @@ def test_agentless(monkeypatch):
     tracer = ddtrace.Tracer()
     info = debug.collect(tracer)
 
-    assert info.get("agent_url", "AGENTLESS")
```
Comment (brettlangdon) [inline tests/integration/test_debug.py:271]: ... so this assert never did anything before?

File: ddtrace/internal/debug.py (hunk 2 above)
Comment (brettlangdon) [reply ddtrace/internal/debug.py:51]: hmm, seeing your test example below.

File: tests/integration/test_debug.py (hunk 3 above)
Comment (Kyle-Verhoog) [reply tests/integration/test_debug.py:271]: yup! 🤦

File: ddtrace/internal/debug.py (hunk 2 above)
Comment (Kyle-Verhoog) [reply ddtrace/internal/debug.py:51]: These logs are just for us/support and in this context we just care about whether an agent is accessible or not which isn't applicable if a custom trace writer is being used.

**Review by jd — APPROVED** [review]

**Pull request merged**